adcirc_nn/coupler/lstmnn.py

```diff
@@ -45,24 +51,56 @@ def initialize(self):
         """Initialize LSTM NN object."""
 
         # Hard-coded values for now:
-        self.dt = 60.0
+        self.dt = 1 # 1 for one hour for now
         self.timer = 0
-        self.niter = 21600
-        self.elbcfunc = lambda t : 5.0e0*(1-np.cos(2.0*np.pi * t / self.tfinal))
+        self.niter = 100
+        # self.elbcfunc = lambda t : 5.0e0*(1-np.cos(2.0*np.pi * t / self.tfinal))
         #self.dummytimes = np.arange(0.0, self.dt*5, self.tfinal)
         #self.dummyvalues = 1.0e3*(1-np.cos(4.0*np.pi/self.dummytimes))
 
-        self.btime = 0.0
         self.tprev = self.timer
         self.tfinal = self.niter
 
+        # load LSTM model
+        nn_input_dir = "nn_input"
+        self.nn_model = self._load_nn_model(nn_input_dir) # provide path if not in the same folder
+        self.hidden = (self.nn_model.c0, self.nn_model.h0)
+
+        # load nn input data
+        df = pd.read_csv(ospathjoin(nn_input_dir, "event.csv"))
+        features = [
+            '43057',
+            '43060',
+            '43053',
+            '43059',
+            '43047',
+            '43058',
+            '43052',
+            '43050',
+            '43160',
+            '43054',
+            '43051',
+            '43049',
+            '43056',
+            '43091',
+            '43055',
+            "Verified (ft)",
+        ]
+        self.features = df[features] # TODO: overwrite this variable during two way coupling
+        assert (self.niter <= len(df)), "lstm model does not have enough input data"
+
     #--------------------------------------------------------------------------#
     def run(self):
-        """Run the LSTM NN object."""
-
-        # Run the NN model. For now, just set the dummy value for next "t"
+        """
+        Run the LSTM NN object.
+        """
         while (self.timer < self.niter):
-            self.elev = self.elbcfunc(self.timer + self.dt)
+            # Time step of NN model
+            # TODO: add scalar transform and inverse transform
+            X = torch.Tensor(self.features.loc[self.timer].values).view(1, 1, -1)
+            y, self.hidden = self.nn_model(X, self.hidden)
+            y = y.view(-1).detach().numpy() # this is the wse prediction at adcirc boundary
+            self.elev = y
```

Comment:
@west5678, AdcircNN needs elevation from NN model in `m`. If I am not wrong, the NN model has been trained on data in `ft`, and I am guessing that the output is in `ft` as well. Could you please check, fix that if needed, and then confirm?

Reply:
I'll fix this. Beyond the SI transformation there is also a data scaling process required by the nn.

Reply:
Okay. Maybe you can add something like a `NN_LENGTH_FACTOR` and `lengthfactor` variables similar to the `NN_TIME_FACTOR` and `timefactor` variables that we have in this class module. We can then use something like `self.elev = y * self.lengthfactor`, maybe.